src/components/organisms/FilterBar.tsx

```diff
@@ -25,14 +27,24 @@ const FilterBar: FC<IProps> = () => {
   const store: RootStore = useStore();
   const classes = useStyles();
   const onFilterChange = useCallback((value: number) => store.changeTimeFilter(value), [store]);
+  const queryParam: string | null = useLocation().search;
```

Comment:
The following lines are hard to read
This entire calculation can be done in eternal function (before the functional component).
a good name might be `extractTimeFilterValueFromQuery`, with a single parameter (the search value).

Also, keep in mind that  `filterValFromUrl` will get a new value in each render, which result in `useEffect` to be invoked in each render. a better approach will be to save the param to a local state (`useState`) and change it only when required. the `useEffect` deps array should use the state value instead of the local variable `filterValFromUrl`.